int32 id
string mode
int32 accel
int32 speed
